libdrgn/program.c

```diff
@@ -185,6 +233,14 @@ drgn_program_set_core_dump(struct drgn_program *prog, const char *path)
 	if (prog->core_fd == -1)
 		return drgn_error_create_os("open", errno, path);
 
+#ifdef LIBKDUMPFILE
+	err = has_kdump_signature(prog->core_fd, &is_kdump);
```

Comment:
How about we:
1. Move `has_kdump_signature()` to `program.c` so it's always available.
2. Add a dummy inline `drgn_program_set_kdump()` in `kdump.h` which is only defined `ifndef LIBKDUMPFILE` which just returns an error explaining that drgn was configured without libkdumpfile.

That way, the user gets a useful error message (and as a bonus we avoid an ifdef in the main logic).